### PR TITLE
Rebase with OE release v0.16.x

### PR DESCRIPTION
**Pull request by mikbras**

### CODE REVIEW

File: prereqs/Makefile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+DRIVER=sgx_linux_x64_driver_1.41.bin
```
Comment (jxyang) [inline prereqs/Makefile:1]: https://github.com/openenclave/openenclave/blob/v0.16.x/docs/GettingStartedDocs/install_oe_sdk-Ubuntu_18.04.md 

Seems OE v0.16 still uses SGX driver 1.35

File: prereqs/Makefile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	sudo apt install -y clang-format-7
+	sudo apt install -y pkg-config
+	sudo apt install -y gdb
+	echo 'deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu bionic main' | sudo tee /etc/apt/sources.list.d/intel-sgx.list
+	wget -qO - https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | sudo apt-key add -
+	echo "deb [arch=amd64] https://packages.microsoft.com/ubuntu/18.04/prod bionic main" | sudo tee /etc/apt/sources.list.d/msprod.list
+	wget -qO - https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -
+	sudo apt update
+	sudo apt install -y libsgx-enclave-common libsgx-quote-ex  libsgx-dcap-ql libsgx-dcap-ql-dev az-dcap-client
+	$(MAKE) install_sgx_driver
```
Comment (jxyang) [inline prereqs/Makefile:22]: Move line 7-15 here so we only need to run `sudo apt update` once.

File: defs.mak [hunk 3]
```diff
@@ -86,13 +86,11 @@ OEENCLAVE_INCLUDES += -I$(TOP)/include
 
 OEENCLAVE_LIBDIR = $(BUILDDIR)/openenclave/lib/openenclave/enclave
 
-OEENCLAVE_LDFLAGS = -nostdlib -nodefaultlibs -nostartfiles -Wl,--no-undefined -Wl,-Bstatic -Wl,-Bsymbolic -Wl,--export-dynamic -Wl,-pie -Wl,--build-id -Wl,-z,noexecstack -Wl,-z,now -Wl,-gc-sections -L$(OEENCLAVE_LIBDIR) -loeenclave -loecryptombed -lmbedtls -lmbedx509 -lmbedcrypto -loelibc -loesyscall -loecore
+OEENCLAVE_LDFLAGS = $(shell PKG_CONFIG_PATH=$(BUILDDIR)/openenclave/share/pkgconfig pkg-config oeenclave-gcc --libs)
 
-OEENCLAVE_CFLAGS_LAX = -g -nostdinc -m64 -fPIE -ftls-model=local-exec -fstack-protector-strong -fno-omit-frame-pointer -ffunction-sections -fdata-sections
+OEENCLAVE_LDFLAGS += $(shell PKG_CONFIG_PATH=$(BUILDDIR)/openenclave/share/pkgconfig pkg-config oeenclave-gcc --variable=mbedtlslibs)
```
Comment (jxyang) [inline defs.mak:91]: I don't think `OEENCLAVE_LDFLAGS` is really needed. Which test case depends on it?

File: tests/gdb/Makefile [hunk 4]
```diff
@@ -31,6 +31,7 @@ tests:
 	$(RUNTEST) $(MAKE) __tests
 
 __tests:
+	${MYST_GDB} ${GDB_CMDS} ${MYST} ${EXEC} rootfs /bin/helloworld ${OPTS}
```
Comment (jxyang) [inline tests/gdb/Makefile:34]: Why do we run the command twice?

File: tools/myst/host/oe_debug_module.h [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+#include <stdint.h>
+
+#define OE_DEBUG_MODULE_VERSION 1
+#define OE_DEBUG_MODULE_MAGIC 0xf67ae6230a18a2ce
+
+typedef struct _debug_module_t
+{
+    uint64_t magic;
+    uint64_t version;
+    struct _debug_module_t* next;
```
Comment (jxyang) [inline tools/myst/host/oe_debug_module.h:18]: Is the `next` pointer needed?

File: tools/myst/config.h [hunk 6]
```diff
@@ -15,7 +15,7 @@
 #define ENCLAVE_STACK_SIZE 8192
 #endif
 
-#define ENCLAVE_HEAP_SIZE (164 * 1024)
+#define ENCLAVE_HEAP_SIZE (256 * 1024)
```
Comment (jxyang) [inline tools/myst/config.h:18]: Is this 256 kb or 256 mb? What caused the increase of the heap size?

File: third_party/openenclave/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+WORKDIR /work/openenclave
+RUN ./scripts/ansible/install-ansible.sh
+RUN ansible-playbook ./scripts/ansible/oe-contributors-acc-setup-no-driver.yml
+
+##
+## Build openenclave
+##
+RUN mkdir build
+WORKDIR /work/openenclave/build
+RUN cmake -DHAS_QUOTE_PROVIDER=OFF -DBUILD_TESTS=OFF -DCMAKE_INSTALL_PREFIX=/work/install ..
```
Comment (bodzhang) [inline third_party/openenclave/Dockerfile:29]: Does the container solution for OE build support OE build switch based on MYST_RELEASE setting?

File: prereqs/Makefile [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+	echo "deb [arch=amd64] https://packages.microsoft.com/ubuntu/18.04/prod bionic main" | sudo tee /etc/apt/sources.list.d/msprod.list
+	wget -qO - https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -
+	sudo apt update
+	sudo apt install -y gdb dkms libssl-dev cryptsetup-bin curl libcurl4-openssl-dev clang-8 clang-format-8 pkg-config python-pip
+	sudo apt install -y libsgx-enclave-common libsgx-quote-ex  libsgx-dcap-ql libsgx-dcap-ql-dev az-dcap-client
+	pip install cmake && sudo ln -f -s $(HOME)/.local/bin/cmake /usr/bin/cmake
+	test -e /dev/sgx || $(MAKE) install_sgx_driver
+
+install_sgx_driver:
+	wget $(DRIVER_URL)/$(DRIVER)
```
Comment (bodzhang) [inline prereqs/Makefile:17]: Certain environment already has DCAP driver or native in-kernel support for SGX. Can the script detect existing SGX support and skip the driver install?

**Comment by mikbras** [comment]
Closing for now

**Pull request closed without merging**